doc/api.md

```diff
@@ -1,230 +1,692 @@
+
+```
+
+#### **API:** GET  /service/attribute
+---
+
+**Parameters:**
+none
+
+**Response:**
+
+code: 
+* 200 -- success
+
+body:
+
+| name | type | description |
+| ---- | ---- | ---------------- |
+| attributes | array | an array of all the attributes for all the services. The fields of a attribute are defined in the following. |
+
+attribute 
+
+| id | string| the id of the attribute. |
+| ---- | ---- | ---------------- |
+| label | string | the user readable name of the attribute |
+| short_type| string | the short type name of the service. This filed is ominted if it is empty. Supported types are: compute, location, architecture and mapped. |
+| sensor_urls | array | an array of sensor url. It applies to all services if it is empty. |
+| publishable| bool | whether the attribute can be made public or not. |
+| mappings | map | a list of key value pairs. |
+
+
+**Example:**
+```
+ curl -s -w %{http_code}  http://localhost/service/attribute | jq  '.'
+{
+  "attributes": [
+    {
+      "id": "app",
+      "sensor_urls": [
+        "https://bluehorizon.network/documentation/netspeed-device-api"
+      ],
+      "label": "app",
+      "publishable": true,
+      "mappings": {
+        "MTN_IS_BANDWIDTH_TEST_ENABLED": "true",
+        "MTN_TARGET_SERVER": "closest"
+      }
+    },
+    {
+      "id": "architecture",
+      "sensor_urls": [
+        "https://bluehorizon.network/documentation/netspeed-device-api"
+      ],
+      "label": "Architecture",
+      "publishable": true,
+      "mappings": {
+        "architecture": "arm"
+      }
+    },
+    {
+      "id": "compute",
+      "sensor_urls": [
+        "https://bluehorizon.network/documentation/netspeed-device-api"
+      ],
+      "label": "Compute Resources",
+      "publishable": true,
+      "mappings": {
+        "cpus": 1,
+        "ram": 0
+      }
+    },
+    {
+      "id": "location",
+      "sensor_urls": [],
+      "label": "Registered Location Facts",
+      "publishable": false,
+      "mappings": {
+        "lat": "41.5861",
+        "lon": "-73.883",
+        "use_gps": false,
+        "user_provided_coords": true
+      }
+    }
+  ]
+}
+
+```
+
+#### **API:** POST  /service/attribute
+---
+
+**Parameters:**
+
+| name | type | description |
+| ---- | ---- | ---------------- |
+| attribute | json | Please refer to the response body for the GET /service/attribute api for the fields of an attribute.  |
+
+**Response:**
+
+code: 
+
+* 200 -- success
+
+body: 
+
+none
+
+**Example:**
+```
+curl -s -w "%{http_code}" -X POST -H 'Content-Type: application/json' -d '{
+    "id": "compute",
+    "short_type": "compute",
+    "label": "Compute Resources",
+    "publishable": true,
+    "mappings": {
+      "ram": 256,
+      "cpus": 1
+    }
+  }'  http://localhost/service/attribute
+
+```
+
+### 4. Agreement
+
+#### **API:** GET  /agreement
+---
+
+**Parameters:**
+none
+
+**Response:**
+
+code: 
+* 200 -- success
+
+body:
+
+| name | type | description |
```

Comment:
There are some new fields just added with my latest PR. Please add protocol_version to this doc. It's the version of the agreement protocol being used.